leakpro/attacks/mia_attacks/lira.py

```diff
@@ -112,36 +147,20 @@ def prepare_attack(self:Self)->None:
             count_in_samples = np.count_nonzero(self.in_indices_mask)
             if count_in_samples > 0:
                 self.logger.info(f"Some shadow model(s) contains {count_in_samples} IN samples in total for the model(s)")
-                self.logger.info("This is not an offline attack!")
+                self.logger.info("This is not an true offline attack!")
 
-        self.skip_indices = np.zeros(len(self.in_indices_mask), dtype=bool)
-        if self.online:
-            no_in = 0
-            no_out = 0
-            for i, mask in enumerate(self.in_indices_mask):
-                if np.count_nonzero(mask) == len(mask):
-                    no_out += 1
-                    self.skip_indices[i] = True
-                elif np.count_nonzero(mask) == 0:
-                    no_in += 1
-                    self.skip_indices[i] = True
-
-            if no_out > 0 or no_in > 0:
-                self.logger.info(f"There are {no_out} audit examples with 0 OUT sample(s) and {no_in} 0 IN sample(s)")
-                self.logger.info("When using few shadow models in online attacks")
-                self.logger.info("some audit sample(s) mighthave a few or even 0 IN or OUT logits")
-                self.logger.info(f"In total {np.count_nonzero(self.skip_indices)} indices will be skipped!")
-
-            if len(self.audit_data) == np.sum(self.skip_indices):
-                raise ValueError("All audit samples are skipped. Please adjust the number of shadow models or the audit dataset.")
-
-        # Calculate logits for all shadow models
         self.logger.info(f"Calculating the logits for all {self.num_shadow_models} shadow models")
-        self.shadow_models_logits = np.swapaxes(np.array(self.signal(self.shadow_models, self.audit_data)), 0, 1)
+        self.shadow_models_logits = np.swapaxes(self.signal(self.shadow_models, self.audit_data), 0, 1)
 
         # Calculate logits for the target model
         self.logger.info("Calculating the logits for the target model")
-        self.target_logits = np.array(self.signal([self.target_model], self.audit_data)).squeeze()
+        self.target_logits = np.swapaxes(self.signal([self.target_model], self.audit_data), 0, 1).squeeze()
+
+        if self.exclude_logit_threshold > 0:
```

Comment:
I dont think exclude_logit_threshold should exist. We should not use audit points without shadow models. Or am I missing something?

Reply:
Removed. Using the convention from RMIA